### PR TITLE
Fixes: [T354304]: Integrate PDL to Commons

### PR DESCRIPTION
**Pull request by okerekechinweotito**
Fixes [T354304](https://phabricator.wikimedia.org/T354304)

## Proposed Changes
- All proposed changes present in #233 apply here (UI Feedbacks , OAuth2 access token and upload to Commons
- For PDL, figure out a way to create a valid PDF file using the zipped images and then upload the PDF to Commons (as that would be easier to use on Wikisource than individually uploading the book page images).

### CODE REVIEW

**Review by coderwassananmol — CHANGES_REQUESTED** [review]
Add JSON.stringify while logging if the variable is an Object

**Comment by coderwassananmol** [comment]
@okerekechinweotito Just tested. Here's the feedback:
1. We are logging and sending highly secure content like OAuth token to Internet Archive which could pose huge security risks. Please do not log and store the OAuth token anywhere.
2. Fields such as Commonsmetadata, Isuploadcommons, Oauthtoken should not be sent to Internet Archive.
3. If the PDL file is of zip type, the file upload to Commons is not working.

Sample IA file with error: https://archive.org/details/KhalsaAkhbarLahore2August1901
Sample PDL file of zip type: http://www.panjabdigilib.org/webuser/searches/displayPage.jsp?ID=682&page=1&CategoryID=5&Searched=

File: bull/commons-queue/consumer.js [hunk 1]
```diff
@@ -1,44 +1,68 @@
...
-      value: null,
-    });
-    return done(null, true);
-    // return done(new Error(`downloadFile: ${downloadFileRes}`));
-  }
-  const commonsResponse = await uploadToCommons(job.data.metadata);
+    if (downloadFileRes.writeFileStatus !== 200) {
+      logger.log({
+        level: "error",
+        message: `downloadFile: ${downloadFileRes}`,
```
[the remark this answers is not in the document]
Comment (okerekechinweotito) [reply bull/commons-queue/consumer.js:41]: @coderwassananmol 
I checked the error log - downloadFileRes is not an object

**Comment by Parthiv-M** [comment]
Couple of things after testing:
- I also find the _Username_ field on the [IA metadata list](https://archive.org/details/werwetcwe). That needs to be removed as well, I figure.
- `{"level":"error","message":"allJobs getJobs TypeError: Cannot read properties of null (reading 'step')","timestamp":"2024-03-19T00:16:22.981Z"}` got this in my logs when upload to commons hits 50%

Rest seems okay

**Comment by okerekechinweotito** [comment]
@Parthiv-M 
I have removed userName

**Comment by coderwassananmol** [comment]
@okerekechinweotito We shouldn't even be logging `oauthtoken` and using or sending it explicitly between components unless **absolutely** required.

**Comment by okerekechinweotito** [comment]
> @okerekechinweotito We shouldn't even be logging `oauthtoken` and using or sending it explicitly between components unless **absolutely** required.
Alright. Please can you have another review

File: utils/helper.js [hunk 2]
```diff
@@ -289,8 +369,10 @@ module.exports = {
       });
 
       const commonsFilePayload = "commonsFilePayload.pdf";
-      let title = metadata.details.volumeInfo.title || metadata.name;
+      let title =
+        metadata.details?.volumeInfo?.title || metadata.name || metadata.title;
       title = title.replaceAll(".", "");
+      console.log(" title:", title);
```
Comment (coderwassananmol) [inline utils/helper.js:375]: Remove console.log statement.

File: bull/pdl-queue/producer.js [hunk 3]
```diff
@@ -26,6 +29,9 @@ module.exports = async (
   metaData["userName"] = userName;
   metaData["IAIdentifier"] = IAIdentifier;
   metaData["isEmailNotification"] = isEmailNotification;
+  metaData["isUploadCommons"] = isUploadCommons;
+  metaData["oauthToken"] = oauthToken;
```
Comment (coderwassananmol) [inline bull/pdl-queue/producer.js:33]: Why is oauthToken required?

File: bull/pdl-queue/producer.js [hunk 4]
```diff
@@ -10,7 +10,10 @@ module.exports = async (
   categoryID,
   email,
   userName,
-  isEmailNotification
+  isEmailNotification,
+  isUploadCommons,
+  oauthToken,
```
Comment (coderwassananmol) [inline bull/pdl-queue/producer.js:15]: Same here.

**Review by coderwassananmol — CHANGES_REQUESTED** [review]

File: utils/helper.js (hunk 2 above)
Comment (okerekechinweotito) [reply utils/helper.js]: @coderwassananmol done

**Review by coderwassananmol — APPROVED** [review]

**Pull request merged**